Mathlib/Combinatorics/SimpleGraph/Triangle/Removal.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Combinatorics.SimpleGraph.DegreeSum
+import Mathlib.Combinatorics.SimpleGraph.Regularity.Lemma
+import Mathlib.Combinatorics.SimpleGraph.Triangle.Basic
+import Mathlib.Combinatorics.SimpleGraph.Triangle.Counting
+
+/-!
+# Triangle removal lemma
+
+In this file, we prove the triangle removal lemma.
+
+## References
+
+[Yaël Dillies, Bhavik Mehta, *Formalising Szemerédi’s Regularity Lemma in Lean*][srl_itp]
+-/
+
+open Finset Fintype Nat SzemerediRegularity
+
+variable {α : Type*} [DecidableEq α] [Fintype α] {G : SimpleGraph α} [DecidableRel G.Adj]
+  {s t : Finset α} {P : Finpartition (univ : Finset α)} {ε : ℝ}
+
+namespace SimpleGraph
+
+/-- An explicit form for the constant in the triangle removal lemma. -/
```

Comment:
bound() here refers to the bound from the regularity lemma, would be good to indicate this in the documentation (in particular point out that the dependence is explicit, but terrible)